common/scripts/gather_packages.py

```diff
@@ -121,27 +121,30 @@ def determineDistTag(branchName, currentVer, latestVer, previousVer):
     packagesToPublish = True
     print ("The package does not yet exist.  Copying " + packageName + " to staging area.")
     shutil.copy(artifact, stagingDir)
-    continue
 
   if 0 != len(serverVer):
     print ("The version already exists.  Skipping...")
     continue
 
   packagesToPublish = True
   print ("Local version is newer than on the server.  Copying package " + packageName + " to staging area.")
-  shutil.copy(artifact, stagingDir)
-
-  command = "npm view " + packageName + " dist-tags.latest"
-  proc = subprocess.Popen(command, stdin = subprocess.PIPE, stdout = subprocess.PIPE, shell=True)
-  latestVer = proc.communicate()[0]
-  if len(latestVer) == 0:
-    print("No version found for dist-tag 'latest'")
+  if not os.path.exists(os.path.join(stagingDir, baseName)):
+    shutil.copy(artifact, stagingDir)
 
-  command = "npm view " + packageName + " dist-tags.previous"
+  command = "npm dist-tag ls " + packageName
   proc = subprocess.Popen(command, stdin = subprocess.PIPE, stdout = subprocess.PIPE, shell=True)
-  previousVer = proc.communicate()[0]
-  if len(previousVer) == 0:
-    print("No version found for dist-tag 'previous'")
+  distTags = proc.communicate()[0]
```

Comment:
We're also figuring out distTags for each package, but we only need to set these values once right?

Should we just do the first package we find that is copied to the staging area? Not sure how much it impacts the runtime

Reply:
added check to see if versions were already set

Reply:
now i have